dask/base.py

```diff
@@ -1088,3 +1095,37 @@ def wait(x, timeout=None, return_when="ALL_COMPLETED"):
         return wait(x, timeout=timeout, return_when=return_when)
     except (ImportError, ValueError):
         return x
+
+
+def clone_key(key):
+    prefix = key_split(key)
+    if isinstance(key, str):
+        return prefix + "-" + tokenize(key)
+    elif isinstance(key, tuple):
+        return (prefix + "-" + tokenize(key[0]), *key[1:])
+    else:
+        raise TypeError(f"Expected a key or tuple key but got {type(key)} instead")
+
+
+def clone(x):
+    """Clone a Dask collection"""
+    from .highlevelgraph import clone_key
+
+    if not is_dask_collection(x):
+        raise TypeError(f"Expected a Dask collection but got {type(x)} instead")
+
+    # Clone the underlying HighLevelGraph
+    dsk = x.__dask_graph__()
+    dsk_new = dsk.clone()
+
+    # Create a new Dask collection from the cloned HighLevelGraph
+    new_name = clone_key(x._name)
+    if is_arraylike(x):
+        from .array.core import new_da_object
+
+        return new_da_object(dsk_new, new_name, x.chunks, x._meta, x.dtype)
+
+    elif is_dataframe_like(x):
+        from .dataframe.core import new_dd_object
+
+        return new_dd_object(dsk_new, new_name, x._meta, x.divisions)
```

Comment:
Maybe this is a good opportunity to make a `new_dask_object` function?  This comes up as well when we convert between different types of dask collections, as in `df.map_partitions(np.asarray)`

Reply:
My alternative proposal here: https://github.com/dask/dask/issues/7093